saleor/static/dashboard/js/components/async-form.js

```diff
@@ -1,25 +1,17 @@
 import {initSelects} from './utils';
 
 export default $(document).on('submit', '.form-async', function (e) {
-  let that = this;
   $.ajax({
-    url: $(that).attr('action'),
+    url: $(e.currentTarget).attr('action'),
     method: 'post',
-    data: $(that).serialize(),
-    complete: function (response) {
-      if (response.status === 400) {
-        $(that).parent().html(response.responseText);
-        initSelects();
-      } else {
-        $('.modal-close').click();
-      }
-    },
-    success: function (response) {
-      if (response.redirectUrl) {
-        window.location.href = response.redirectUrl;
-      } else {
-        location.reload();
-      }
+    data: $(e.currentTarget).serialize(),
+    success: function (data) {
+      const $message = $(data).find('div');
+      $message.find('a').remove();
+      $(e.currentTarget).find('.row').last().html($message);
+      $(e.currentTarget).find('.row').last().css('text-align', 'center');
+      $(e.currentTarget).find('svg').height('160px');
+      $(e.currentTarget).find('button').remove();
```

Comment:
Please don't re-evaluate the same expressions over and over again, `$(...)` is not free and neither is `.find(...)`.